test/e2e/e2e_duplicatepods_test.go

```diff
@@ -0,0 +1,198 @@
+			replicasNum: 4,
+			beforeFunc: func(deployment *appsv1.Deployment) {
+				deployment.Spec.Replicas = func(i int32) *int32 { return &i }(4)
+				deployment.Spec.Template.Spec.NodeName = workerNodes[0].Name
+			},
+			expectedEvictedPodCount: 2,
+		},
+		{
+			description: "Evict Pod even Pods with local storage",
+			replicasNum: 5,
+			beforeFunc: func(deployment *appsv1.Deployment) {
+				deployment.Spec.Replicas = func(i int32) *int32 { return &i }(5)
+				deployment.Spec.Template.Spec.Volumes = []v1.Volume{
+					{
+						Name: "sample",
+						VolumeSource: v1.VolumeSource{
+							EmptyDir: &v1.EmptyDirVolumeSource{
+								SizeLimit: resource.NewQuantity(int64(10), resource.BinarySI)},
+						},
+					},
+				}
+			},
+			expectedEvictedPodCount: 2,
+		},
+	}
+	for _, tc := range tests {
+		t.Logf("Creating deployment %v in %v namespace", deploymentObj.Name, deploymentObj.Namespace)
+		tc.beforeFunc(deploymentObj)
+
+		_, err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).Create(ctx, deploymentObj, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("Error creating deployment: %v", err)
+			if err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"app": "test-duplicate", "name": "test-duplicatePods"})).String(),
+			}); err != nil {
+				t.Fatalf("Unable to delete deployment: %v", err)
+			}
+			return
+		}
+		//defer clientSet.AppsV1().Deployments(deploymentObj.Namespace).Delete(ctx, deploymentObj.Name, metav1.DeleteOptions{})
+		waitForPodsRunning(ctx, t, clientSet, map[string]string{"app": "test-duplicate", "name": "test-duplicatePods"}, tc.replicasNum, testNamespace.Name)
+
+		// Run DeschedulerStrategy strategy
+		evictionPolicyGroupVersion, err := eutils.SupportEviction(clientSet)
+		if err != nil || len(evictionPolicyGroupVersion) == 0 {
+			t.Fatalf("Error creating eviction policy group %v", err)
+		}
+		podEvictor := evictions.NewPodEvictor(
+			clientSet,
+			evictionPolicyGroupVersion,
+			false,
+			0,
+			nodes,
+			true,
+			false,
+			false,
+		)
+
+		t.Log("Running DeschedulerStrategy strategy")
+		strategies.RemoveDuplicatePods(
+			ctx,
+			clientSet,
+			deschedulerapi.DeschedulerStrategy{
+				Enabled: true,
+				Params: &deschedulerapi.StrategyParameters{
+					RemoveDuplicates: &deschedulerapi.RemoveDuplicates{},
+				},
+			},
+			workerNodes,
+			podEvictor,
+		)
+
+		waitForTerminatingPodsToDisappear(ctx, t, clientSet, testNamespace.Name)
+		actualEvictedPodCount := podEvictor.TotalEvicted()
+		if actualEvictedPodCount != tc.expectedEvictedPodCount {
+			t.Errorf("Unexpected number of pods have been evicted, got %v, expected %v", actualEvictedPodCount, tc.expectedEvictedPodCount)
+		}
+
+		clientSet.AppsV1().Deployments(deploymentObj.Namespace).Delete(ctx, deploymentObj.Name, metav1.DeleteOptions{})
```

Comment:
better to do this up top and use defer

Reply:
Fix done. Would you please kindly review?